libindy_vdr/src/pool/networker/zmq.rs

```diff
@@ -750,6 +759,39 @@ impl RemoteNode {
         } else {
             debug!("Socks proxy is not configured");
         }
+
+        // Try to bind to a client port within the configured range,
+        // fallback to default behavior if all available ports are in use or no range is set
+        if let Some((min_port, max_port)) = client_port_range {
+            let mut port_bound = false;
+            for port in min_port..=max_port {
+                let bind_addr = format!("tcp://*:{}", port);
+                match s.bind(&bind_addr) {
+                    Ok(_) => {
+                        debug!("Bound ZMQ client socket to port {}", port);
+                        port_bound = true;
+                        break;
+                    }
+                    Err(zmq::Error::EADDRINUSE) => {
+                        // Port is in use, try the next one
+                        continue;
+                    }
+                    Err(err) => {
+                        // Other error, log and continue
+                        warn!("Failed to bind client socket to port {}: {}", port, err);
+                        continue;
+                    }
+                }
+            }
+
+            if !port_bound {
+                warn!(
+                    "Could not bind to any port in range {}-{}, using default ZMQ behavior",
+                    min_port, max_port
+                );
+            }
+        }
```

Comment:
The min and max ports taken from env vars allows us to narrow the ports, potentially avoiding any firewall issues that exist with the default non-standard ports